File: pe-downloader.ml
open Batteries

let p_from = Sys.argv.(1) |> Int.of_string
let p_to = Sys.argv.(2) |> Int.of_string
let url = "http://projecteuler.net/problem="

let f l = let tmp = Unix.system @@ "wget -k -p -e robots=off "^url^(String.of_int l) in ()

let () =
  List.iter f @@ List.range p_from `To p_to;
  Sys.rename "projecteuler.net" ("Problems"^Sys.argv.(1)^"-"^Sys.argv.(2))

File: pe-downloader.py
#!/usr/bin/env python3

import subprocess,os

url = "http://projecteuler.net/problem="
args = ['wget','-k','-p','-e','robots=off']
problems_from = 101
problems_to = 150

for i in range(problems_from,problems_to+1):
    subprocess.call(args+[url+str(i)])

os.rename("projecteuler.net","Problems"+str(problems_from)+"-"+str(problems_to))
